Jenkinsfile

```diff
@@ -1,150 +1,175 @@
+
+            stage('Static Analysis') {
+            steps {
+                sh './gradlew clean pmd checkstyle lint'
+            }
+
+            post {
+                always {
+                    pmd         canComputeNew: false, canRunOnFailed: true, defaultEncoding: '', healthy: '', pattern: "${env.GRADLE_PROJECT_MODULE_NAME}/build/reports/pmd.xml",        unHealthy: '', unstableTotalAll: '0'
+                    checkstyle  canComputeNew: false, canRunOnFailed: true, defaultEncoding: '', healthy: '', pattern: "${env.GRADLE_PROJECT_MODULE_NAME}/build/reports/checkstyle.xml", unHealthy: '', unstableTotalAll: '0'
+                    androidLint canComputeNew: false, canRunOnFailed: true, defaultEncoding: '', healthy: '', pattern: "${env.GRADLE_PROJECT_MODULE_NAME}/build/reports/lint*.xml",      unHealthy: '', unstableTotalAll: '0'
+                }
+            }
+        }
+
+        stage('Unit and Device tests') {
+            steps {
+                // Run local unit tests
+                sh './gradlew -PenableCoverage -Pjenkins clean jacocoTestDebugUnitTestReport'
+                // Convert the JaCoCo coverate to the Cobertura XML file format.
+                // This is done since the Jenkins JaCoCo plugin does not work well.
+                // See also JENKINS-212 on jira.catrob.at
+                sh "if [ -f '$JACOCO_UNIT_XML' ]; then ./buildScripts/cover2cover.py $JACOCO_UNIT_XML > $JAVA_SRC/coverage1.xml; fi"
+                // ensure that the following test run does not overwrite the results
+                sh "mv ${env.GRADLE_PROJECT_MODULE_NAME}/build ${env.GRADLE_PROJECT_MODULE_NAME}/build-unittest"
+
+                // Run device tests
+                sh './gradlew -PenableCoverage -Pjenkins clean startEmulator adbDisableAnimationsGlobally createDebugCoverageReport'
+                // Convert the JaCoCo coverate to the Cobertura XML file format.
+                // This is done since the Jenkins JaCoCo plugin does not work well.
+                // See also JENKINS-212 on jira.catrob.at
+                sh "if [ -f '$JACOCO_XML' ]; then ./buildScripts/cover2cover.py $JACOCO_XML > $JAVA_SRC/coverage2.xml; fi"
+            }
+
+            post {
+                always {
+                    junit '**/*TEST*.xml'
+                    step([$class: 'CoberturaPublisher', autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: "$JAVA_SRC/coverage*.xml", failUnhealthy: false, failUnstable: false, maxNumberOfBuilds: 0, onlyStable: false, sourceEncoding: 'ASCII', zoomCoverageChart: false, failNoReports: false])
+                    
+                    sh './gradlew stopEmulator clearAvdStore'
+                    archiveArtifacts 'logcat.txt'  
+                    
+                    plot csvFileName: 'dexcount.csv',
+                        csvSeries: [[displayTableFlag: false,
+                        exclusionValues: '',
+                        file: 'Paintroid/build/outputs/dexcount/*.csv',
```

Comment:
Just in-case.
The build-directory does not survive a ./gradlew clean.
Furthermore the build-directory is not kept for consecutive builds.
Not sure if this is needed here to have previous results.

Reply:
The build-directory does not have to be persistent to plot the previous results.
Should we not move to spaces instead of tabs ? 
We did the same on catroid with the hardwaretests